web-app/js/smartR/_angular/services/smartRUtils.js

```diff
@@ -19,8 +19,8 @@ window.smartRApp.factory('smartRUtils', ['$q', function($q) {
         return String(str).replace(/[^a-z0-9]/g, function(s) {
             var c = s.charCodeAt(0);
             if (c == 32) return '-';
-            if (c >= 65 && c <= 90) return s.toLowerCase();
-            return ('000' + c.toString(16)).slice(-4);
+            if (c >= 65 && c <= 90) return '_' + s.toLowerCase();
+            return '__' + ('000' + c.toString(16)).slice(-4);
```

Comment:
This method makes CSS safe versions of every given string. There is less magic going on than it looks like.
In words: "look for everything that is not a-z and 0-9 and replace it with '-' if it's a space, lowerCase if it's upperCase and with the hexcode if it's anything else." 

The prefix _ and __ are simply to avoid any collision and indicate that a concersion (i.e. into hexcode) happened.